orttraining/orttraining/training_ops/cuda/activation/bias_gelu_grad_impl.cu

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "orttraining/training_ops/cuda/activation/bias_gelu_grad_impl.h"
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "core/providers/cuda/shared_inc/fast_divmod.h"
+#include "orttraining/training_ops/cpu/activation/gelu_computation_mode.h"
+#include "orttraining/training_ops/cuda/activation/gelu_grad_impl_common.cuh"
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T, typename GeluComputationMode, int num_consecutive_elements_per_group, int num_groups_per_thread>
+__global__ void BiasGeluGradDxKernel(
+    CUDA_LONG input_size, fast_divmod bias_size_fdm,
+    const T* dY, const T* X, const T* B, T* dX) {
+  const auto& num_threads_per_block = blockDim.x;
+  const int& bias_size = bias_size_fdm.d_;
+  CUDA_LONG base_idx =
+      num_consecutive_elements_per_group * num_groups_per_thread * num_threads_per_block * blockIdx.x +
+      num_consecutive_elements_per_group * threadIdx.x;
+
+#pragma unroll
```

Comment:
This is effectively handling 16 elements per thread... 

Would it make a perf difference if we just pass 16 elementsPerThread with a stride...

Reply:
yes, there are some measurements in a reply to another comment. it does a bit worse.

---
In reply to: [441114083](https://github.com/microsoft/onnxruntime/pull/4146#discussion_r441114083) [](ancestors = 441114083)